compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -125,7 +125,7 @@ impl<K: Hash + Eq, V> interpret::AllocMap<K, V> for FxIndexMap<K, V> {
     where
         K: Borrow<Q>,
     {
-        FxIndexMap::remove(self, k)
+        FxIndexMap::swap_remove(self, k)
```

Comment:
This feels pre-existing, but it feels odd to me that we're using an IndexMap (presumably for stable iteration order?) but we're calling swap_remove here. I think we may want to either do the same deprecation in favor of an explicit swap_remove (and consider fallout)...

Or perhaps we could just use a regular map where we don't the index properties :)

Probably just file an issue and add a TODO here since this is not something we should fix in a PR like this.

Same applies in a few other cases where we flipped to swap_remove.

Reply:
Yes in the version of `indexmap` currently in use, `remove` just forwards to `swap_remove` so no behaviour change.
Since there's a lot of places where `remove` is being used (and is probably not what we really want to be doing?) I'm not sure of a good place for a fixme.

Reply:
I think perhaps we can #[allow(deprecated)] in as targeted a manner as reasonable and then file an issue with the description and a link to this PR as "how to find".

Reply:
Or a single-line comment on all the sites added here, something like `// FIXME(#XXX): consider if swap_removed is the right option`.

Reply:
I've filed #120456, I'll add some fixme comments pointing at it